### PR TITLE
docs - Hive profile supports column projection and pushdown

### PR DESCRIPTION
**Pull request by lisakowen**
misc doc updates for the recent Hive* work.  in this PR:
- add a "choosing the profile" topic to main hdfs page that mentions which data types have more than one profile, and some criteria for choosing
- add info about pxf.ppd.hive property in pxf-site.xml
- add column projection info
- add filter pushdown info - I AM NOT SURE IF I AM REPRESENTING THIS CORRECTLY IN THE TABLE
- hive topic - discuss new PPD option, remove limitation.  

note:  wanted to get this part out for review; i am still determining if other edits are required to the hive topic.

doc review site links:
- choosing the profile - http://docs-lisa-hive-colppd.cfapps.io/pxf/5-16/using/access_hdfs.html#choose_profile
- pxf.ppd.hive property description - http://docs-lisa-hive-colppd.cfapps.io/pxf/5-16/using/cfg_server.html#pxf-site
- filter pushdown - http://docs-lisa-hive-colppd.cfapps.io/pxf/5-16/using/filter_push.html
- hive topic PPD - http://docs-lisa-hive-colppd.cfapps.io/pxf/5-16/using/hive_pxf.html#hive_queryextdata

### CODE REVIEW

File: docs/content/filter_push.html.md.erb [hunk 1]
```diff
@@ -57,8 +57,10 @@ PXF accesses data sources using profiles exposed by different connectors, and fi
 | s3:parquet and s3:text with S3-Select | Y |  N | Y | Y | Y | Y | Y |
 | HBase | Y | N | Y | N | Y | Y | N |
 | Hive | Y<sup>2</sup> | N | N | N | Y<sup>2</sup> | Y<sup>2</sup> | N |
+| Hive (accessing stored as Parquet) | Y, Y<sup>2</sup> | N | N | Y | Y, Y<sup>2</sup> | Y, Y<sup>2</sup> | Y |
+| Hive (accessing stored as RCFile or ORC) | Y, Y<sup>2</sup> | N | Y | Y | Y, Y<sup>2</sup> | Y, Y<sup>2</sup> | Y |
```
Comment (frankgh) [inline docs/content/filter_push.html.md.erb:61]: I think line should be merged with line 59,  but let's confirm with Alex.

File: docs/content/access_hdfs.html.md.erb [hunk 2]
```diff
@@ -119,7 +119,25 @@ The PXF Hadoop connectors expose the following profiles to read, and in many cas
...
 
-You provide the profile name when you specify the `pxf` protocol on a `CREATE EXTERNAL TABLE` command to create a Greenplum Database external table that references a Hadoop file, directory, or table. For example, the following command creates an external table that uses the default server and specifies the profile named `hdfs:text`:
+
+### <a id="choose_profile"></a>Choosing the Profile
+
+PXF provides more than one profile to access text and Parquet data on Hadoop. Here are some things to consider as you determine which profile to choose.
+
+Choose the `Hive` profile when:
+
+- The data resides in a Hive table, and you do not know the type of data up front.
```
Comment (frankgh) [inline docs/content/access_hdfs.html.md.erb:129]:
```suggestion
- The data resides in a Hive table, and you do not know the underlying file type of the table up front.
```

**Review by frankgh — COMMENTED** [review]
I have some comments, but we might need Alex's input

File: docs/content/col_project.html.md.erb [hunk 3]
```diff
@@ -11,7 +11,7 @@ Column projection is automatically enabled for the `pxf` external table protocol
 | Data Source | Connector | Profile(s) |
 |-------------|---------------|---------|
 | External SQL database | JDBC Connector | Jdbc |
-| Hive | Hive Connector | Hive, HiveRC, HiveORC, HiveVectorizedORC |
+| Hive | Hive Connector | Hive (accessing tables stored as Parquet, RCFile, and ORC), HiveRC, HiveORC, HiveVectorizedORC |
```
Comment (denalex) [inline docs/content/col_project.html.md.erb:14]: also for tables stored as text files

File: docs/content/filter_push.html.md.erb (hunk 1 above)
Comment (denalex) [reply docs/content/filter_push.html.md.erb:61]: this is tricky -- Hive itself does not make sense, the pushdown is different based on the file storage format for a table partition, each partition can have different pushdown eve when accessed by the same Hive profile. I'd remove Hive line altogether. Also `HiveText` does not support any pushdown (other than partition pruning). And I think VectorizedORC does not support pushdown at all.

**Review by denalex — COMMENTED** [review]
Looks good. At some point we need to go over the details again and maybe re-phrase partition filter pushdown to partition pruning, as the 2 concepts, while similar are implemented differently.

File: docs/content/filter_push.html.md.erb (hunk 1 above)
Comment (frankgh) [reply docs/content/filter_push.html.md.erb:61]: I checked vectorized ORC and we don't do predicate pushdown, just partition pruning. We should be able to use the sarg for vectorized ORC.

File: docs/content/access_hdfs.html.md.erb [hunk 4]
```diff
@@ -119,7 +119,25 @@ The PXF Hadoop connectors expose the following profiles to read, and in many cas
...
+### <a id="choose_profile"></a>Choosing the Profile
+
+PXF provides more than one profile to access text and Parquet data on Hadoop. Here are some things to consider as you determine which profile to choose.
+
+Choose the `Hive` profile when:
+
+- The data resides in a Hive table, and you do not know the underlying file type of the table up front.
+- The data resides in a Hive table, and the Hive table is partitioned.
+
+Choose the `hdfs:text` or `hdfs:parquet` profile when:
```
Comment (dyozie) [inline docs/content/access_hdfs.html.md.erb:132]: Mixing both of these in the intro sentences was a little confusing to me.  Doesn't the first bullet always go with hdfs:text and the second one with hdfss:parquet?  If so I'd just make each a separate paragraph "Choose the hdfs:text profile when you know the file type and you know the location of the file in the HDFS file system." "Choose the hdfs:parquet profile when the file e is Parquet, you know ..., "

File: docs/content/access_hdfs.html.md.erb [hunk 5]
```diff
@@ -119,7 +119,25 @@ The PXF Hadoop connectors expose the following profiles to read, and in many cas
...
+
+- You know the file type, and you know the location of the file in the HDFS file system.
+- The file is Parquet, you know the location of the file in the HDFS file system, and you want to take advantage of extended filter pushdown support for additional data types and operators.
+
+
+### <a id="specify_profile"></a>Specifying the Profile
+
+You must provide the profile name when you specify the `pxf` protocol on a `CREATE EXTERNAL TABLE` command to create a Greenplum Database external table that references a Hadoop file or directory, or HBase or Hive table. For example, the following command creates an external table that uses the default server and specifies the profile named `hdfs:text` to access the HDFS file `/data/pxf_examples/pxf_hdfs_simple.txt`:
```
Comment (dyozie) [inline docs/content/access_hdfs.html.md.erb:140]: on a -> in a 
, or HBase or Hive table -> , HBase table, or Hive table

**Review by dyozie — APPROVED** [review]

**Pull request merged**